minichain/p2p.py

```
@@ -385,6 +402,21 @@ async def broadcast_block(self, block):
+        logger.info("Network: Broadcasting chain request")
+        payload = {"type": "chain_request", "data": {}}
+        await self._broadcast_raw(payload)
+
+    async def send_chain_response(self, blocks_dicts, writer):
+        logger.info("Network: Sending chain response with %d blocks", len(blocks_dicts))
+        payload = {"type": "chain_response", "data": {"blocks": blocks_dicts}}
+        line = (canonical_json_dumps(payload) + "\n").encode()
+        try:
+            writer.write(line)
+            await writer.drain()
+        except Exception as e:
```

Ruff (0.15.17)

[warning] Do not catch blind exception: `Exception`

(BLE001)